### PR TITLE
[Test] rerouting tests for 4.x

### PR DESCRIPTION
**Pull request by IHEII**
### Task Description

Add tests for testing rerouting in 4.x observer.

### Solution Description

Add tests for testing rerouting in 4.x observer.

### Passed Regressions

Pass all tests except index tests.

### Upgrade Compatibility

N.A.

### Other Information

Rerouting tests should be run manually. 

### Release Note

N.A.

### CODE REVIEW

File: test/reroute/util/replica_operation.go [hunk 1]
```diff
@@ -30,7 +31,10 @@ import (
 )
 
 const (
-	getReplicaSql = "SELECT A.table_id as table_id, A.partition_id as partition_id, A.svr_ip as svr_ip, B.svr_port as svr_port, A.role as role FROM oceanbase.__all_virtual_proxy_schema A inner join oceanbase.__all_server B on A.svr_ip = B.svr_ip and A.sql_port = B.inner_port WHERE tenant_name = ? and database_name = ? and table_name = ? and partition_id in "
+	getReplicaSql      = "SELECT A.table_id as table_id, A.partition_id as partition_id, A.svr_ip as svr_ip, B.svr_port as svr_port, A.role as role FROM oceanbase.__all_virtual_proxy_schema A inner join oceanbase.__all_server B on A.svr_ip = B.svr_ip and A.sql_port = B.inner_port WHERE tenant_name = ? and database_name = ? and table_name = ? and partition_id in "
+	getLDIdSql         = "SELECT ls_id FROM cdb_ob_table_locations WHERE tenant_id = 1 and database_name = ? and table_name = ? and role = 'LEADER'"
+	getFollowerAddrSql = "SELECT concat(svr_ip,':',svr_port) AS host FROM __all_virtual_ls_meta_table WHERE tenant_id = 1 and ls_id = ? and role = 2 and replica_status = 'NORMAL' limit 1;"
```
Comment (WeiXinChan) [inline test/reroute/util/replica_operation.go:36]: use oceanbase database

File: protocol/move_replica_info.go [hunk 2]
```diff
@@ -154,7 +154,11 @@ func (i *ObTableMoveReplicaInfo) Decode(buffer *bytes.Buffer) {
 	i.ObUniVersionHeader.Decode(buffer)
 	i.tableId = uint64(util.DecodeVi64(buffer))
 	i.schemaVersion = uint64(util.DecodeVi64(buffer))
-	i.partitionId = uint64(util.DecodeVi64(buffer))
+	if util.ObVersion() >= 4 {
```
[the remark this answers is not in the document]
Comment (IHEII) [reply protocol/move_replica_info.go:161]: Fixed

File: test/reroute/util/replica_operation.go (hunk 1 above)
Comment (IHEII) [reply test/reroute/util/replica_operation.go]: Fixed

**Comment by WeiXinChan** [comment]
LGTM

**Pull request merged**